winter_messaging_transactional/rabbitmq/topology_configurator.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from collections import defaultdict
+from typing import Dict
+
+from injector import inject
+
+from winter.core import Component
+from winter.messaging import EventHandlerRegistry
+from winter.messaging import get_event_topic
+from winter.messaging import TopicAnnotation
+from winter.messaging import MessagingConfig
+from winter_messaging_transactional.rabbitmq.rabbitmq_client import RabbitMQClient
+from winter_messaging_transactional.naming_convention import get_consumer_queue
+from winter_messaging_transactional.naming_convention import get_exchange_name
+from winter_messaging_transactional.naming_convention import get_routing_key
+
+
+log = logging.getLogger(__name__)
+
+
+class InvalidTopologyException(Exception):
+    pass
+
+
+class TopologyConfigurator:
+    @inject
+    def __init__(
+        self,
+        rabbit_client: RabbitMQClient,
+        handler_registry: EventHandlerRegistry,
+        config: MessagingConfig
+    ) -> None:
+        self._rabbit_client = rabbit_client
+        self._handler_registry = handler_registry
+        self._config = config
+        self._topic_to_exchange_map: Dict[str, str] = {}
+        self._consumer_id_to_queue = {}
+        self._consumer_to_events_map = defaultdict(set)
+        self._configure()
+
+    def _configure(self):
+        for consumer_id, packages in self._config.consumers.items():
```

Comment:
@mikhaillazko Why we iterate over all of consumers from `MessagingConfig`? Why we don't filter them by `consumer_id` from args?

Reply:
because we could not create partially topology. it's centralized things and alternative is setup it manually and check it by eyes that your infrastructure topology the same as configuration. Pretty similar what celery do when start worker